$MirOS: ports/net/centericq/patches/patch-src_hooks_abstracthook_cc,v 1.1 2006/10/29 02:45:12 tg Exp $
--- src/hooks/abstracthook.cc.orig	Sun Dec 23 21:49:27 2007
+++ src/hooks/abstracthook.cc	Sun Dec 23 21:49:33 2007
@@ -46,7 +46,9 @@ time_t timer_current = time(0);
 
 abstracthook::abstracthook(protocolname aproto)
     : proto(aproto), searchdest(0)
-{ }
+{
+	protocharset = NULL;
+}
 
 void abstracthook::init() {
 }
@@ -233,6 +235,18 @@ string abstracthook::ruscrlfconv(const s
 
 string abstracthook::rusconv(const string &tdir, const string &text) {
     string r;
+
+    if (conf.getisunicode()) {
+#ifdef HAVE_ICONV
+	if (protocharset != NULL) {
+		if (tdir == "kw" || tdir == "ku")
+			return siconv(text, "utf-8", protocharset);
+		if (tdir == "wk" || tdir == "uk")
+			return siconv(text, protocharset, "utf-8");
+	}
+#endif
+	return text;
+    }
 
     if(!conf.getcpconvert(proto) && tdir.find("u") == -1)
 	return text;
